content/best_practices.adoc

```diff
@@ -56,7 +56,48 @@ Some operations require logging/auditing to allow to understand what happened du
+
+Sometimes, when not having a lot of interactions with user on UI, (like for Wallet & Perkstore widgets), it's more relevant to compute the HTML to display in Server-End using a simple JSP file without having to mount a Vue application for it and to retrieve its information using additional REST calls. Using Vue applications is relevant only when :
+
+* the computed information to display is slow: in this case, using a REST call to asynchronously retrieve information from Server-End will allow to detach heavy computing from critical path of Web-Page DOM retrieving
+* there are complex UI components and user interactions in application DOM
+
+As a conclusion, to ensure having a good Front-End performances when developing a Vue application, we have multiple techniques that can be applied, but there is no exact coding pattern that we can follow. Each application has its specifities and performances requirement has to be considered in conception phase, else you can looase time to refactor code. In order to help developers to continuously measure developed/maintained applications performances, a tooling has been added to display application performances in Browser console when the server is started in https://docs.exoplatform.org/en/6.1/GetStarted.html#dev-mode[Dev Mode]. To enable this, use:
+
+* `Vue.createApp` to create a new Vue application instead of `new Vue`
+* use statement `this.$root.$applicationLoaded()` when all Data is fetched and the UI has been displayed to the End-user.
+
+=== REST API Performances
+
+By increasing the number of REST calls in a single Web-Page, this will allow to retrieve information in parallel. This was a major evolution of eXo Platform product architecture introduced in version 6 which will reduce the classic Server-End stateful DOM rendering (that was made using JSP, Servlet, Portlet, JSF, Spring MVC...). As an advantage of this, the Rendering phase is decentralized and is deported on Client-Side. Consequently, the Server will hold less information in its memory (Stateful UI Tree) and make less computing related to page rendering. As a significant drawback of this new architecture, making more parallel computing (parallel REST calls for each page) makes the multi-threading aspect more important to consider. In fact, a REST call has to be performant and must not hold any potential thread-blocking statement. In addition, the HTTP cache headers has to be considered for almost all REST calls. We have several mime types retrieved through REST calls:
```

Comment:
```suggestion

* `Vue.createApp` to create a new Vue application instead of `new Vue`
* use statement `this.$root.$applicationLoaded()` when all Data is fetched and the UI has been displayed to the End-user.

=== REST API Performances

By increasing the number of REST calls in a single Web-Page, will allow retrieval of information in parallel. This was a major evolution of eXo Platform product architecture introduced in version 6 which will reduce the classic Server-End stateful DOM rendering (that was made using JSP, Servlet, Portlet, JSF, Spring MVC...). As an advantage of this, the Rendering phase is decentralized and is deported on the Client-Side. Consequently, the Server will hold less information in its memory (Stateful UI Tree) and make less computing related to page rendering. As a significant drawback of this new architecture, making more parallel computing (parallel REST calls for each page) makes the multi-threading aspect more important to consider. In fact, a REST call has to be performant and must not hold any potential thread-blocking statement. In addition, the HTTP cache headers have to be considered for almost all REST calls. We have several mime types retrieved through REST calls:
```